api/v1alpha1/hostedcluster_types.go

```diff
@@ -71,6 +71,8 @@ type HostedClusterSpec struct {
 	// Defaults to SingleReplica when not set.
 	// +optional
 	ControllerAvailabilityPolicy AvailabilityPolicy `json:"controllerAvailabilityPolicy,omitempty"`
+
+	Etcd EtcdSpec `json:"etcd"`
```

Comment:
@csrwng @enxebre @sjenning @relyt0925 

Details I think we should sort out from this point downward before merging:

1. Docs on fields
2. API metadata tagging for optionality, enum types, etc
3. How to handle defaulting

Reply:
docs added

Reply:
can we may be default this Etcd struct to managed similar to how we do for https://github.com/openshift/hypershift/blob/1230dd4706ff8a394645f0976ced121318ddfeba/api/v1alpha1/nodepool_types.go#L62-L64

Reply:
We can! doing it!